src/radical/pilot/agent/executing/popen.py

```diff
@@ -762,7 +767,10 @@ def _get_exec(self, task, launcher):
         ret  = ''
 
         for cmd in ru.as_list(launcher.get_exec(task)):
-            ret += '%s\n' % cmd
+            ret += '%s &\n' % cmd
+            # collect rank PID
+            ret += 'RP_RANK_PID=$!\n'
+            ret += 'wait $RP_RANK_PID\n'
```

Comment:
Assume `get_exec` returns a list of commands - now the pid for the *last* command will be stored in `RP_RANK_PID` - is that what we want?

Reply:
Empty string is fine - as long it is quoted the shell is ok with it, and we need to quote anyway.  So I would be in favor of that if that works for you.  But I don't feel strongly about it, it's a minor nitpick...